smartpy/optimizers/sgd.py

```diff
@@ -0,0 +1,51 @@
+import theano
+import theano.tensor as T
+from collections import OrderedDict
+
+import numpy as np
+from . import Optimizer
+
+
+class SGD(Optimizer):
+    def __init__(self, model, loss_fct, dataset, batch_size=1):
+        super(SGD, self).__init__(model, loss_fct, dataset)
```

Comment:
Add a `Loss` object that builds the graph and can be interrogated to get the inputs variables.